### PR TITLE
BF: allow for int type string in ls on BE begin with <i

### LINKED ISSUE #422: nib_ls test failure on SPARC
@yarikoptic - any thoughts on this one?

http://nipy.bic.berkeley.edu/builders/nibabel-py2.7-wheezy-sparc/builds/115

```
======================================================================
FAIL: nibabel.tests.test_scripts.test_nib_ls_multiple
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/buildslave/nd-bb-slave-sparc64-wheezy/nibabel-py2_7-wheezy-sparc/build/venv/local/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/buildslave/nd-bb-slave-sparc64-wheezy/nibabel-py2_7-wheezy-sparc/build/venv/local/lib/python2.7/site-packages/nibabel/tests/test_scripts.py", line 94, in test_nib_ls_multiple
    "Full output was: %s" % stdout_lines)
AssertionError: Type sub-string didn't start with 'i'. Full output was: [u'/home/buildslave/nd-bb-slave-sparc64-wheezy/nibabel-py2_7-wheezy-sparc/build/venv/local/lib/python2.7/site-packages/nibabel/tests/data/example4d.nii.gz      <i2 [128,  96,  24,   2] 2.00x2.00x2.20x2000.00  #exts: 2 sform', u'/home/buildslave/nd-bb-slave-sparc64-wheezy/nibabel-py2_7-wheezy-sparc/build/venv/local/lib/python2.7/site-packages/nibabel/tests/data/example_nifti2.nii.gz <i2 [ 32,  20,  12,   2] 2.00x2.00x2.20x2000.00  #exts: 2 sform', u'/home/buildslave/nd-bb-slave-sparc64-wheezy/nibabel-py2_7-wheezy-sparc/build/venv/local/lib/python2.7/site-packages/nibabel/tests/data/small.mnc             <i2 [ 18,  28,  29]      9.00x8.00x7.00', u'/home/buildslave/nd-bb-slave-sparc64-wheezy/nibabel-py2_7-wheezy-sparc/build/venv/local/lib/python2.7/site-packages/nibabel/tests/data/nifti2.hdr            <i2 [ 91, 109,  91]      2.00x2.00x2.00']
```

### PR DESCRIPTION
**Pull request by yarikoptic**
Closes  #422

### CODE REVIEW

**Comment by matthew-brett** [comment]
Is it obvious why we're not getting this failure on the other sparc or the PowerPC buildbot?

**Comment by yarikoptic** [comment]
ah... may be now you will? ;-)

**Comment by yarikoptic** [comment]
appveyor errors seems to be unrelated
```
Error: MD5 sums mismatch for download: http://repo.continuum.io/pkgs/free/win-32/libtiff-4.0.6-vc10_1.tar.bz2 (39eec3f8aad0d28a6e4e69127682f5e9 != e631280d4c50b860362611e2db3468f4)
Command exited with code 1
```

File: nibabel/tests/test_scripts.py [hunk 1]
```diff
@@ -97,7 +98,9 @@ def test_nib_ls_multiple():
 
     # they should be indented correctly.  Since all files are int type -
     ln = max(len(f) for f in fnames)
-    assert_equal([l[ln:ln+2] for l in stdout_lines], [' i']*4,
+    little_endian = sys.byteorder == 'little'
```
Comment (matthew-brett) [inline nibabel/tests/test_scripts.py]: How about:

```python
i_str = ' i' if sys.byteorder == 'little' else ' <i'
assert_equal([l[ln:ln + len(i_str)] for l in stdout_lines], [i_str] * 4,
      msg="Type sub-string didn't start with '%s'. "
                "Full output was: %s" % (i_str, stdout_lines))
```

Comment (yarikoptic) [inline nibabel/tests/test_scripts.py]: sure, why not ;) will push now

**Comment by matthew-brett** [comment]
Thanks - sorry for the a-hole micro-comments - too much time on my hands.

**Comment by yarikoptic** [comment]
Exactly what I thought ;-)

**Pull request merged**